planner/core/planbuilder.go

```diff
@@ -1256,7 +1256,7 @@ func filterPathByIsolationRead(ctx sessionctx.Context, paths []*util.AccessPath,
 	if dbName.L == mysql.SystemDB {
 		return paths, nil
 	}
-	isolationReadEngines := ctx.GetSessionVars().GetIsolationReadEngines()
+	isolationReadEngines := GetAvailableIsolationReadEngines4Plan(ctx.GetSessionVars())
```

Comment:
Only here in the optimize phase need the `isolationReadEngines`. But it's not easy to get the `ast`. So we can not directly use `IsReadOnly` function to check. So we store the `IsReadyOnlyStmt` in advance.